test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/Binders/CollectionModelBinderProviderTest.cs

```diff
@@ -66,46 +66,11 @@ public void Create_ForSupportedTypes_ReturnsBinder(Type modelType)
             Assert.IsType<CollectionModelBinder<int>>(result);
         }
 
-        // These aren't ICollection<> - we can handle them by creating a List<> - but in this case
-        // we can't set the property so we can't bind.
-        [Theory]
-        [InlineData(nameof(ReadOnlyProperties.Enumerable))]
-        [InlineData(nameof(ReadOnlyProperties.ReadOnlyCollection))]
-        [InlineData(nameof(ReadOnlyProperties.ReadOnlyList))]
```

Comment:
Suggest adding these types to the data for `Create_ForSupportedTypes_ReturnsBinder()`.